### ISSUE TITLE
Compilation hangs at XmlMarshaller.cpp

### ISSUE THREAD

**Issue by michal-devel** [trufanov-nok/scantailor-universal#111]
Hi!
I'm having trouble compiling the code. The compilation has always failed when reached the file XmlMarshaller.cpp. At this point my computer has always freezen so I needed to hard reset it. I use g++ (GCC) 12.1.0 on Arch Linux. This problem doesn't occur when compiling ScanTailor from https://github.com/vigri/scantailor-advanced. I can give more information if needed.

Thanks for your work on ST!

**Comment by trufanov-nok** [comment]
Hi!
>  this point my computer has always freezen so I needed to hard reset it.

That's really strange. Are you compiling in virtual box or on some cloud machine? Could you run `make 2> ~/log.txt` and check the ~/log.txt after that?

**Comment by michal-devel** [comment]
No, I use a a desktop with Linux as my OS. Please find below my exact build commands (taken from Arch package):

```
build() {
    export LDFLAGS="-L/usr/local/lib,--rpath=/usr/local/lib"
    export LD_LIBRARY_PATH="/usr/local/lib"
    export CFLAGS="-fPIC"
    export CXXFLAGS="-fPIC"

    cd "${srcdir}/${pkgname%-git}"
    sed -i "s/#define VERSION .*$/#define VERSION \"$pkgver\"/" version.h
    # https://aur.archlinux.org/packages/scantailor-advanced-git/#comment-831404
    mkdir build && cd build
    cmake -G "Unix Makefiles" ..
    cmake --build . --parallel ${nproc}
    make -j${nproc}
}
```

I will run the command you mentioned and share the log afterwards.

**Comment by michal-devel** [comment]
This is strange as it worked flawlessly this time. I skipped `cmake --build . --parallel ${nproc}` and executed instead make. No crash, no error. Just a quick compilation! Thanks for you assistance!
